Description: fix FTBFS with clang instead of gcc
Author: Nicolas Sévelin-Radiguet <nicosr@free.fr>
Last-Update: 2014-04-20

--- a/tools/ppp/ppp.c++
+++ b/tools/ppp/ppp.c++
@@ -627,7 +627,7 @@
 //
 //////////////////////////////////////////////////////////////////////////////
 
-main(int, char *argv[])
+int main(int, char *argv[])
 {
     char	buf[1024];
     Line	line;
--- a/apps/tools/ivcat/ivcat.c++
+++ b/apps/tools/ivcat/ivcat.c++
@@ -215,7 +215,7 @@
     }
 }
 
-main(int argc, char **argv)
+int main(int argc, char **argv)
 {
     int expandFileNodes = 0;
     int expandTextureNodes = 0;
--- a/apps/tools/ivfix/ivfix.c++
+++ b/apps/tools/ivfix/ivfix.c++
@@ -80,7 +80,7 @@
 //
 /////////////////////////////////////////////////////////////////////////////
 
-main(int argc, char **argv)
+int main(int argc, char **argv)
 {
     OptionInfo options;
 
--- a/apps/tools/ivinfo/ivinfo.c++
+++ b/apps/tools/ivinfo/ivinfo.c++
@@ -159,7 +159,7 @@
     }
 }
 
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
     SoInput	in;
     SoNode	*root;
