--- latencytop-0.5.orig/fsync.c
+++ latencytop-0.5/fsync.c
@@ -51,7 +51,7 @@
 static GList *fsync_data;
 
 
-static chain_file(struct fsync_process *proc, char *filename)
+static void chain_file(struct fsync_process *proc, char *filename)
 {
 	struct fsync_files *file;
 	GList *item;
@@ -75,7 +75,7 @@
 	proc->files = g_list_append(proc->files, file);
 }
 
-static report_file(char *process, char *file)
+static void report_file(char *process, char *file)
 {
 	struct fsync_process *proc;
 	GList *item;
