Description: fix FTBFS with clang instead of gcc
Author: Nicolas Sévelin-Radiguet <nicosr@free.fr>
Last-Update: 2014-03-26

--- a/s2s/out.c
+++ b/s2s/out.c
@@ -627,7 +627,7 @@
             nad_free(pkt->nad);
         free(pkt);
 
-        return;
+        return 0;
     }
 
     /* new route key */
--- a/sm/mod_roster.c
+++ b/sm/mod_roster.c
@@ -307,7 +307,7 @@
     pkt_sess(push, rw->sess);
 }
 
-static void _roster_set_item(pkt_t pkt, int elem, sess_t sess, mod_instance_t mi)
+static _roster_set_item(pkt_t pkt, int elem, sess_t sess, mod_instance_t mi)
 {
     mod_roster_t mroster = (mod_roster_t) mi->mod->private;
     module_t mod = mi->mod;
@@ -322,7 +322,7 @@
     jid = jid_new(NAD_AVAL(pkt->nad, attr), NAD_AVAL_L(pkt->nad, attr));
     if(jid == NULL) {
         log_debug(ZONE, "jid failed prep check, skipping");
-        return;
+        return 0;
     }
 
     /* check for removals */
@@ -378,7 +378,7 @@
 
         jid_free(jid);
 
-        return;
+        return 0;
     }
 
     /* find a pre-existing one */
@@ -393,7 +393,7 @@
 
             /* if the limit is reached, skip it */
             if (ret == st_SUCCESS && items >= mroster->maxitems)
-                return;
+                return 0;
         }
 
         /* make a new one */
